vdr/api/v1/api.go

```diff
@@ -56,132 +58,131 @@ func (a *Wrapper) ResolveStatusCode(err error) int {
 	})
 }
 
-// Preprocess is called just before the API operation itself is invoked.
-func (a *Wrapper) Preprocess(operationID string, context echo.Context) {
-	context.Set(core.StatusCodeResolverContextKey, a)
-	context.Set(core.OperationIDContextKey, operationID)
-	context.Set(core.ModuleNameContextKey, "VDR")
-}
-
 // DeleteVerificationMethod accepts a DID and a KeyIdentifier of a verificationMethod and calls the DocManipulator
 // to remove the verificationMethod from the given document.
-func (a *Wrapper) DeleteVerificationMethod(ctx echo.Context, didStr string, kidStr string) error {
-	id, err := did.ParseDID(didStr)
+func (a *Wrapper) DeleteVerificationMethod(_ context.Context, request DeleteVerificationMethodRequestObject) (DeleteVerificationMethodResponseObject, error) {
+	id, err := did.ParseDID(request.Did)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	kid, err := did.ParseDIDURL(kidStr)
+	kid, err := did.ParseDIDURL(request.Kid)
 	if err != nil {
-		return core.InvalidInputError("given kid could not be parsed: %w", err)
+		return nil, core.InvalidInputError("given kid could not be parsed: %w", err)
 	}
 
 	err = a.DocManipulator.RemoveVerificationMethod(*id, *kid)
 	if err != nil {
-		return fmt.Errorf("could not remove verification method from document: %w", err)
+		return nil, fmt.Errorf("could not remove verification method from document: %w", err)
 	}
-	return ctx.NoContent(http.StatusNoContent)
+	return DeleteVerificationMethod204Response{}, nil
 }
 
 // AddNewVerificationMethod accepts a DID and adds a new VerificationMethod to that corresponding document.
-func (a *Wrapper) AddNewVerificationMethod(ctx echo.Context, id string) error {
-	d, err := did.ParseDID(id)
+func (a *Wrapper) AddNewVerificationMethod(_ context.Context, request AddNewVerificationMethodRequestObject) (AddNewVerificationMethodResponseObject, error) {
+	d, err := did.ParseDID(request.Did)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	req := AddNewVerificationMethodJSONRequestBody{}
-	if err := ctx.Bind(&req); err != nil {
-		return err
+
+	opts := request.Body
+	if opts == nil {
+		opts = &VerificationMethodRelationship{}
 	}
 
-	vm, err := a.DocManipulator.AddVerificationMethod(*d, req.ToFlags(vdrDoc.DefaultCreationOptions().KeyFlags))
+	vm, err := a.DocManipulator.AddVerificationMethod(*d, opts.ToFlags(vdrDoc.DefaultCreationOptions().KeyFlags))
 	if err != nil {
-		return err
+		return nil, err
 	}
-	return ctx.JSON(http.StatusOK, *vm)
+	return AddNewVerificationMethod200JSONResponse(*vm), nil
 }
 
 func (a *Wrapper) Routes(router core.EchoRouter) {
-	RegisterHandlers(router, a)
+	RegisterHandlers(router, NewStrictHandler(a, []StrictMiddlewareFunc{
```

Comment:
move this one to the top and add audit routing?

Reply:
Fixed (had to rebase due to conflicts from this)